boa_parser/src/parser/statement/mod.rs

```diff
@@ -283,6 +294,25 @@ where
             let item =
                 StatementListItem::new(self.allow_yield, self.allow_await, self.allow_return)
                     .parse(cursor, interner)?;
+
+            if directive_prologues {
+                if let ast::StatementListItem::Statement(ast::Statement::Expression(
+                    ast::Expression::Literal(ast::expression::literal::Literal::String(string)),
+                )) = &item
+                {
+                    if interner.resolve_expect(*string).join(
+                        |s| s == "use strict",
+                        |g| g == utf16!("use strict"),
+                        true,
+                    ) {
+                        cursor.set_strict_mode(true);
+                        strict = true;
+                    }
+                } else {
+                    directive_prologues = false;
+                }
+            }
```

Comment:
I think I found a small bug. On this PR, this will throw:

```javascript
function f(){
    "a";
    ;
    "use strict";
    let public = 5;
}
```

However, I ran that on Spidermonkey and it successfully runs:

![image](https://user-images.githubusercontent.com/38230983/211948871-62f6479d-11d1-4ece-9815-cc4beed56f8c.png)

This means the parser should stop considering string expressions as directives if there's an empty statement between them.